federation/pkg/federation-controller/replicaset/planner/planner_test.go

```diff
@@ -28,8 +28,33 @@ func doCheck(t *testing.T, pref map[string]fed_api.ClusterReplicaSetPreferences,
 	planer := NewPlanner(&fed_api.FederatedReplicaSetPreferences{
 		Clusters: pref,
 	})
-	plan := planer.Plan(replicas, clusters)
+	plan, overflow := planer.Plan(replicas, clusters, map[string]int64{}, map[string]int64{})
 	assert.EqualValues(t, expected, plan)
+	assert.Equal(t, 0, len(overflow))
+}
+
+func doCheckWithExisting(t *testing.T, pref map[string]fed_api.ClusterReplicaSetPreferences, replicas int64, clusters []string,
+	existing map[string]int64, expected map[string]int64) {
+	planer := NewPlanner(&fed_api.FederatedReplicaSetPreferences{
+		Clusters: pref,
+	})
+	plan, overflow := planer.Plan(replicas, clusters, existing, map[string]int64{})
+	assert.Equal(t, 0, len(overflow))
+	assert.EqualValues(t, expected, plan)
```

Comment:
Does EqualValues work correctly for maps? I would really suggest using reflect,DeepEqual as everywhere in the codebase.